Server/ReasnAPI/ReasnAPI/Controllers/EventsController.cs

```diff
@@ -1,171 +1,307 @@
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using ReasnAPI.Models.Enums;
+using ReasnAPI.Services;
+using FluentValidation;
+using Microsoft.AspNetCore.Http.Extensions;
+using ReasnAPI.Mappers;
+using ReasnAPI.Models.API;
 
 namespace ReasnAPI.Controllers;
 
 [ApiController]
 [Route("[controller]")]
-public class EventsController : ControllerBase
+public class EventsController(
+    ReasnContext context,
+    EventService eventService,
+    UserService userService,
+    ImageService imageService,
+    AddressService addressService,
+    TagService tagService)
+    : ControllerBase
 {
-    private readonly ReasnContext _context;
 
-    public EventsController(ReasnContext context)
-    {
-        _context = context;
-    }
-
-    [HttpGet]
+    [HttpGet]
+    [ProducesResponseType<IEnumerable<EventResponse>>(StatusCodes.Status200OK)]
     public IActionResult GetEvents()
     {
-        throw new NotImplementedException();
+        var events = eventService.GetAllEvents();
+   
+        return Ok(events);
     }
 
     [HttpPost]
     [Authorize(Roles = "Admin, Organizer")]
-    public IActionResult CreateEvent()
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    public IActionResult CreateEvent([FromBody] EventCreateRequest eventRequest, [FromServices] IValidator<EventDto> validator)
     {
-        throw new NotImplementedException();
+        var user = userService.GetCurrentUser();
+        
+        var eventDto = eventRequest.ToDto(user.Id);
+        eventDto.Slug = "temp";
```

Comment:
Can't we assure that will always be at least an empty string if it is not nullable? Or change the validator as we (you) implemetned slug creating in eventservice?

Reply:
slug is nullable but validators requiers it to proper validate so i just created temp slag that is overriten later in create method